File: src/test/java/com/wxl/crawlerdytt/es/EsClientTest.java
package com.wxl.crawlerdytt.es;

import org.elasticsearch.action.search.SearchRequest;
import org.elasticsearch.action.search.SearchResponse;
import org.elasticsearch.client.RequestOptions;
import org.elasticsearch.client.RestHighLevelClient;
import org.elasticsearch.search.SearchHit;
import org.elasticsearch.search.aggregations.AggregationBuilders;
import org.elasticsearch.search.aggregations.Aggregations;
import org.elasticsearch.search.aggregations.bucket.terms.Terms;
import org.elasticsearch.search.aggregations.bucket.terms.TermsAggregationBuilder;
import org.elasticsearch.search.builder.SearchSourceBuilder;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.test.context.junit4.SpringRunner;

import java.util.List;

/**
 * Create by wuxingle on 2020/7/12
 */
@RunWith(SpringRunner.class)
@SpringBootTest
public class EsClientTest {

    @Autowired
    private RestHighLevelClient client;

    @Test
    public void test1() throws Exception {
        SearchSourceBuilder sourceBuilder = new SearchSourceBuilder();
        TermsAggregationBuilder aggregationBuilder = AggregationBuilders.terms("years").field("year");
        sourceBuilder.aggregation(aggregationBuilder);
        sourceBuilder.size(0);

        SearchRequest request = new SearchRequest();
        request.indices("dytt");
        request.types("_doc");
        request.source(sourceBuilder);

        SearchResponse response = client.search(request, RequestOptions.DEFAULT);
        System.out.println(response);

        Aggregations aggregations = response.getAggregations();

        System.out.println(aggregations.asMap());

        Terms years = aggregations.get("years");

        System.out.println("-----------------------------------");
        List<? extends Terms.Bucket> buckets = years.getBuckets();
        for (Terms.Bucket bucket : buckets) {
            System.out.println(bucket.getKeyAsString() + ":" + bucket.getDocCount());
        }
    }
}

File: src/test/kotlin/com/wxl/dyttcrawler/es/EsClientTest.kt
package com.wxl.dyttcrawler.es

import org.elasticsearch.action.search.SearchRequest
import org.elasticsearch.client.RequestOptions
import org.elasticsearch.client.RestHighLevelClient
import org.elasticsearch.search.aggregations.AggregationBuilders
import org.elasticsearch.search.aggregations.bucket.terms.Terms
import org.elasticsearch.search.builder.SearchSourceBuilder
import org.junit.jupiter.api.Test
import org.junit.jupiter.api.extension.ExtendWith
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.test.context.junit.jupiter.SpringExtension

/**
 * Create by wuxingle on 2021/10/12
 *
 */
@ExtendWith(SpringExtension::class)
@SpringBootTest
class EsClientTest {

    @Autowired
    lateinit var client: RestHighLevelClient

    @Test
    fun test1() {
        val sourceBuilder = SearchSourceBuilder()
        val aggregationBuilder = AggregationBuilders.terms("years").field("year")
        sourceBuilder.aggregation(aggregationBuilder)
        sourceBuilder.size(0)

        val request = SearchRequest()
        request.indices("dytt")
        request.source(sourceBuilder)

        val response = client.search(request, RequestOptions.DEFAULT)
        println(response)

        val agg = response.aggregations
        println(agg.asMap())

        val years = agg.get<Terms>("years")
        println("----------------------------")

        val buckets = years.buckets
        for (bucket in buckets) {
            println(bucket.keyAsString + ":" + bucket.docCount)
        }
    }


}